src/game/resources.rs

```diff
@@ -1 +1,77 @@
+use bevy::prelude::*;
+use rand::{seq::SliceRandom, thread_rng};
 
+use super::WordComplexity;
+
+/// Collection of words to choose from at "random"
+///
+/// * Must always use as ResMut<WordBank> due to modification for word uniqueness
+#[derive(Resource)]
+pub struct WordBank {
+    pub easy: Vec<String>,
+    easy_ptr: usize,
+    pub med: Vec<String>,
+    med_ptr: usize,
+    pub hard: Vec<String>,
+    hard_ptr: usize,
+}
+pub trait RandomWord {
+    fn get_word(&mut self, complexity: WordComplexity) -> String;
+}
+
+impl RandomWord for WordBank {
+    fn get_word(&mut self, complexity: WordComplexity) -> String {
+        match complexity {
+            WordComplexity::Easy => self.get_easy_word(),
+            WordComplexity::Medium => self.get_med_word(),
+            WordComplexity::Hard => self.get_hard_word(),
+        }
+    }
+}
+
+// Vecs will be shuffled on creation
+impl WordBank {
```

Comment:
While unlikely, still has the chance of producing a word already on the screen, so we should at least check for that somehow

Reply:
The chances of that will be super low (would need to be on screen, hit end of vec, reshuffle, then get randomly chosen again) especially as we add more words. That may be something to look into once we have our first level setup. I'm not sure how I would be checking what words are on screen yet

Reply:
Also may be worth having the caller deal with that because if you're trying to assign a random word to an enemy, you already will have an enemy query

Reply:
ya its a weird edge case but an obvious one that *could* happen and *would* break the game to a degree. I like the reshuffle on level complete option, though if in theory we get through all the words in level, it could still happen, if say, you dont use your special. Think we may just need to query for all words, and check them against the random one, so maybe leaving it up to the caller is fine, though we likely should build a shared function to handle most of that